File: 1539.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int findKthPositive(vector<int>& arr, int k) {
        int last = 0;
        for(int n : arr) {
            if(n - last > 1) {
                k -= n - last - 1;
                last = n;
                if(k <= 0) { last--; break; }
            }else {
                last = n;
            }
        }
        return last + k;
    }
};

File: src/1539.py
class Solution:
    def findKthPositive(self, arr: list[int], k: int) -> int:
        last = 0
        for n in arr:
            if n - last > 1:
                k -= n - last - 1
                last = n
                if k <= 0:
                    last -= 1
                    break
            else:
                last = n
        return last + k
